wordpress-mobile/WordPress-iOS
# [Home Page Picker] Analytic Events

**Fixes** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2723

**Related PR**
- `WordPress-iOS-Shared`: https://github.com/wordpress-mobile/WordPress-iOS-Shared/pull/279

📓 This PR implements the suggested approach mentioned here: https://github.com/wordpress-mobile/gutenberg-mobile/issues/2723#issuecomment-723246016
📓This is built off of https://github.com/wordpress-mobile/WordPress-iOS/pull/15231 so this build can be used to test the changes there as well.

## To test:
<details>
<summary><strong>To disable or enable the development version of Home Page Picker</strong></summary>

- Open the app from the build that allows FeatureFlags such as a PR build or a local development build
    - By default, the modal layout picker will be disabled.
    - From the site page:
        - Click on your Gravatar.
        - Click on App Settings.
        - Click on Debug.
        - Toggle "Site Creation: Home Page Picker" to enable or disable the picker
        - <kdb><a href="https://user-images.githubusercontent.com/3384451/97343011-5f032980-185d-11eb-9030-06c17f732fec.gif"><img width="300" src="https://user-images.githubusercontent.com/3384451/97343011-5f032980-185d-11eb-9030-06c17f732fec.gif"/></a></kdb>

</details>

When enabled, the first step of the Site Creation flow will be replaced with the Home Page Picker flow.

For these test cases in addition to verifying in our internal Analytics tool you can validate the event by filtering the console logs on `🔵 Tracked:` 

### enhanced_site_creation_site_design_viewed
1. Navigate to the Create WordPress.com site
    - My Sites > ➕ > Create WordPress.com site
**Expect** to see `enhanced_site_creation_site_design_viewed` fire when the screen loads

### enhanced_site_creation_site_design_selected
1. Navigate to the Create WordPress.com site
    - My Sites > ➕ > Create WordPress.com site
1. Select a Design
1. Tap on "Choose"
**Expect** to see `enhanced_site_creation_site_design_selected ` fire with the property `template` filled out with the `slug` for the design.

### enhanced_site_creation_site_design_skipped
1. Navigate to the Create WordPress.com site
    - My Sites > ➕ > Create WordPress.com site
1. Tap on "Skip"
**Expect** to see `enhanced_site_creation_site_design_skipped` fire.

### error
1. Disconnect your device from the network
1. Navigate to the Create WordPress.com site
    - My Sites > ➕ > Create WordPress.com site
**Expect** to see `enhanced_site_creation_error_shown` with `error_info` populated with a description of the error that occurred.

## Additional Context: 

This is a continuation of the work done in the [Home Page Picker Project](https://github.com/wordpress-mobile/gutenberg-mobile/labels/Project%3A%20Home%20Page%20Picker). 

## PR submission checklist:
